tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -3016,6 +3101,115 @@ public function userHasStoredEtagOfElementOnPathFromSpace($user, $path, $storePa
 		}
 	}
 
+	/**
+	 * Request to lock the resource inside of space
+	 *
+	 * @param string $user
+	 * @param string $resource
+	 * @param TableNode $properties
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function sendRequestToLockResouceInsideOfSpace(string $user, string $resource, TableNode $properties, string $spaceName):void {
+		$body
+			= "<?xml version='1.0' encoding='UTF-8'?>" .
+			"<d:lockinfo xmlns:d='DAV:'> ";
+		$headers = [];
+		$this->featureContext->verifyTableNodeRows($properties, [], ['lockscope', 'depth', 'timeout']);
+		$propertiesRows = $properties->getRowsHash();
+		foreach ($propertiesRows as $property => $value) {
+			if ($property === "depth" || $property === "timeout") {
+				//properties that are set in the header not in the xml
+				$headers[$property] = $value;
+			} else {
+				$body .= "<d:$property><d:$value/></d:$property>";
+			}
+		}
+		$body .= "</d:lockinfo>";
+		$space = $this->getSpaceByName($user, $spaceName);
+		$fullUrl = $space['root']['webDavUrl'] . '/' . ltrim($resource, '/');
+		$this->featureContext->setResponse(
+			HttpRequestHelper::sendRequest(
+				$fullUrl,
+				"",
+				'LOCK',
+				$user,
+				$this->featureContext->getPasswordForUser($user),
+				[],
+				$body
+			)
+		);
+		$this->featureContext->theHTTPStatusCodeShouldBe(
+			200,
+			__METHOD__ . " Failed to lock the resource $resource"
+		);
+		$responseXml = $this->featureContext->getResponseXml(null, __METHOD__);
+		$this->featureContext->setResponseXmlObject($responseXml);
+		$xmlPart = $responseXml->xpath("//d:locktoken/d:href");
+		if (\is_array($xmlPart) && isset($xmlPart[0])) {
+			$this->tokenOfLastLock[$user][$resource] = (string) $xmlPart[0];
+		} else {
+			Assert::fail("could not find lock token after trying to lock '$resource'");
+		}
+	}
+
+    /**
+     * @Then /^as user "([^"]*)" the lock discovery property "([^"]*)" of the (?:file|folder|entry) "([^"]*)" inside space "([^"]*)" should match "([^"]*)"$/
+     *
+     * @param string|null $user
+     * @param string $xpath
+     * @param string $path
+     * @param string $spaceName
+     * @param string $pattern
+     *
+     * @return void
+     * @throws Exception
+     */
+    public function userGetsPropertiesOfFolderOfSpaceAndAssertValueOfItemInResponseToUserRegExp(string $user, string $xpath, string $path, string $spaceName, string $pattern):void {
+        $space = $this->getSpaceByName($user, $spaceName);
+        $fullUrl = $space['root']['webDavUrl'] . '/' . ltrim($path, '/');
+        $propertiesTable = new TableNode([['propertyName'],['d:lockdiscovery']]);
+        $properties = null;
+        $this->featureContext->verifyTableNodeColumns($propertiesTable, ["propertyName"]);
+        $this->featureContext->verifyTableNodeColumnsCount($propertiesTable, 1);
```

Comment:
the table is self-created, so validations are of no use they will always pass
also, a table would be better for these long arguments IMO.
that would also help to check for multiple resources.